newIDE/app/src/EventsSheet/EventsTree/Instruction.js

```diff
@@ -239,6 +243,43 @@ const Instruction = (props: Props) => {
                 .getPlainString();
               expressionIsValid = resourcesManager.hasResource(resourceName);
             }
+            if (
+              expressionIsValid &&
+              parameterType === 'functionParameterName'
+            ) {
+              const eventsFunction = props.scope.eventsFunction;
+              if (eventsFunction) {
+                const eventsBasedEntity =
+                  props.scope.eventsBasedBehavior ||
+                  props.scope.eventsBasedObject;
+                const functionsContainer = eventsBasedEntity
+                  ? eventsBasedEntity.getEventsFunctions()
+                  : props.scope.eventsFunctionsExtension;
+
+                if (functionsContainer) {
+                  const allowedParameterTypes = parameterMetadata
+                    .getExtraInfo()
+                    .split(',');
+                  const parameters = enumerateParametersUsableInExpressions(
+                    functionsContainer,
+                    eventsFunction,
+                    allowedParameterTypes
+                  );
+                  const functionParameterNameExpression = instruction
+                    .getParameter(parameterIndex)
+                    .getPlainString();
+                  const functionParameterName = functionParameterNameExpression.substring(
+                    1,
+                    functionParameterNameExpression.length - 1
+                  );
+                  expressionIsValid = parameters.some(
+                    parameter => parameter.getName() === functionParameterName
+                  );
+                }
+              } else {
+                expressionIsValid = false;
```

Comment:
Can you add a comment to tell that this case should only happen if an instruction using a functionParameterName was used in a scope where it's not even relevant (which would be a UX fail)?

Reply:
Users could change the type of a parameter after writing some events.

Reply:
Users could copy-paste events from an extension.